src/library/scala/collection/immutable/StringLike.scala

```diff
@@ -66,8 +66,9 @@ self =>
   /** Return the current string concatenated `n` times.
    */
   def * (n: Int): String = {
+    val str = toString
     val buf = new StringBuilder
-    for (i <- 0 until n) buf append toString
+    for (i <- 0 until n) buf append str
```

Comment:
Might be worth decompiling this (you can do it in intellij) and see what kind of code it is generating.